src/snowflake/connector/cursor.py

```diff
@@ -76,16 +77,16 @@
     from .connection import SnowflakeConnection
     from .file_transfer_agent import SnowflakeProgressPercentage
     from .result_batch import ResultBatch
+    from pyarrow import Table
 
```

Comment:
if we directly add it here, I think it would break users who do not have pyarrow installed in their environment.

Reply:
It shouldn't, since it's under the `if TYPE_CHECKING` conditional.  I have just now reordered the imports so that the import from pyarrow is next to the import from pandas, so as to make this more clear. Pandas is another package that is not a dependency of snowflake without extras installed (and both pyarrow and pandas are installed by the `[pandas]` extra).

These imports are only tried by the type checker, not at runtime. The usage of the imported classes in type hints in the function definitions is allowed due to the presence of this import at the top of the file: `from __future__ import annotations`. You can read more about Postponed evaluation of annotations [here on stackoverflow](https://stackoverflow.com/a/61544901/1018861). This feature is already being used for [pandas in this file here](https://github.com/snowflakedb/snowflake-connector-python/blob/990df441805f91907248beb9bef1d06cbc6c8f04/src/snowflake/connector/cursor.py#L1360).

You can also try running the following code block to convince yourself. It won't throw an error about `FooBar` (unless you explicitly call `main()`, but that also applies to calling `fetch_pandas_all` without pandas installed).

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from import_that_doesnt_exist import FooBar

def main() -> FooBar:
    return FooBar()

print("Hello, world!") # this will run
```

Reply:
Thanks. Testing the PR here: https://github.com/snowflakedb/snowflake-connector-python/pull/1838